tests/reframe/testlib/applications/gromacs/__init__.py

```diff
@@ -0,0 +1,67 @@
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+# 3.5.0 Required for using current_partition.processor
+#@rfm.required_version('>=3.5.0')
+class Gromacs(rfm.RunOnlyRegressionTest, pin_prefix=True):
+    '''Gromacs benchmark based on Prace Benchmark Suite GROMACS case A.
+
+    Derived tests must specify the variables ``num_tasks``, ``num_tasks_per_node``, ``num_cpus_per_task``, ``nsteps`` and ``modules``.
+    Note that a sufficiently large ``nsteps`` needs to be defined in order for GROMACS to pass the load balancing phase.
+    As a rough estimate: 10000 steps would generally be ok for 24 tasks, a 100000 steps for 240 tasks, etc.
+    '''
+
+    num_tasks = required
+    num_tasks_per_node = required
+    num_cpus_per_task = required
+    nsteps = required
```

Comment:
This variable needs to be declared first (`num_tasks`, `num_tasks_per_node` and so on were declared in the `rfm.RegressionTest` class already.)
```suggestion
    nsteps = variable(int)
```
I've created an issue (https://github.com/eth-cscs/reframe/issues/2011) to raise an error when this is attempted.

Reply:
Ah, I see! I indeed copied what you did for `num_tasks` in https://github.com/eth-cscs/reframe/pull/1996/ , but failed to realize that this was actually a special case since the `variable` was already declared as part of the `ReFrame` framework. Cool, ok, I'll replace by `variable` to get the type checking :)